althea_kernel_interface/src/stats.rs

```diff
@@ -0,0 +1,88 @@
+use super::{KernelInterface, KernelInterfaceError};
+
+use std::collections::HashMap;
+
+use failure::Error;
+use std::fs::File;
+use std::io::Read;
+use regex::Regex;
+
+impl KernelInterface {
+    pub fn get_proc_stat(&self) -> Result<String, Error> {
+        let re = Regex::new(r"(?m)^([a-z0-9]+)(?:\s([a-z0-9]+))+$").unwrap();
+        let mut f = File::open("/proc/stat")?;
+
+        let mut contents = String::new();
+        f.read_to_string(&mut contents)?;
+
```

Comment:
does this need to stick around?

Reply:
Assuming you meant the comment of the outdated code, that is now removed